internal/config/config_file.go

```diff
@@ -51,30 +52,84 @@ func ConfigDir() string {
 func autoMigrateConfigDir(newPath string) {
 	path, err := os.UserHomeDir()
 	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
-		migrateConfigDir(oldPath, newPath)
+		migrateDir(oldPath, newPath)
 		return
 	}
 
 	path, err = homedir.Dir()
 	if oldPath := filepath.Join(path, ".config", "gh"); err == nil && dirExists(oldPath) {
-		migrateConfigDir(oldPath, newPath)
+		migrateDir(oldPath, newPath)
 	}
 }
 
-func dirExists(path string) bool {
-	f, err := os.Stat(path)
-	return err == nil && f.IsDir()
+func StateDir() string {
+	if path := os.Getenv(XDG_STATE_HOME); path != "" {
+		path = filepath.Join(path, "gh")
+		if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
+			_ = os.MkdirAll(path, 0755)
+			autoMigrateStateDir(path)
+		}
+		return path
+	}
+
+	return ConfigDir()
```

Comment:
Super-duper nitpick: would we consider an alternate default for app state if XDG_STATE_HOME isn't set? For example, we now moved the config to `AppData` on Windows. However, AppData is synced across machines. It's not ideal that the state file is synced, since it could lead to race conditions. We should use `LocalAppData` for things that shouldn't be synced.

For now, we only use the state file for remembering that we've done a version check, but we might use it for other things in the future. Just a thought

Reply:
I am completely open to having an alternate fallback for the state directory. I choose `ConfigDir()` just because it was the smallest behavioral change since that is the directory we were using previously to store state. 

On Windows we can use `LocalAppData` and then on Mac/Linux we can use `$HOME/.local/state` as the XDG spec specified. Additionally we probably would want to introduce a `GH_STATE_DIR` to match `GH_CONFIG_DIR`. Does that make sense @mislav? @vilmibm Do you have an opinion on this?

Reply:
I don't feel as strongly for having GH_STATE_DIR as I felt for GH_CONFIG_DIR. Switching GH_CONFIG_DIR has use-cases, but I cannot envision a use-case for intentionally switching GH_STATE_DIR. People can use the XDG approach if they wish to modify it.

Reply:
I also don't think we need our own specific GH_STATE_DIR for now; if someone has a pressing need for it in the future we can support it.

I would prefer the alternate fallback, though-- `LocalAppData` and `$HOME/.local.state` makes sense to me.